core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/search/InSearchOperatorDsl.kt

```diff
@@ -0,0 +1,125 @@
+package com.github.inflab.spring.data.mongodb.core.aggregation.search
+
+import com.github.inflab.spring.data.mongodb.core.annotation.AggregationMarker
+import org.bson.Document
+import org.bson.types.ObjectId
+import java.time.temporal.Temporal
+
+/**
+ * A Kotlin DSL to configure the in operator using idiomatic Kotlin code.
+ *
+ * @author cranemont
+ * @since 1.0
+ * @see <a href="https://www.mongodb.com/docs/atlas/atlas-search/in/#in">in</a>
+ */
+@AggregationMarker
+class InSearchOperatorDsl {
+    private val document = Document()
+
+    /**
+     * The indexed field to search.
+     * You can also specify a wildcard path to search.
+     * See path construction for more information.
+     *
+     * @param path Indexed field to search.
+     * @see <a href="https://www.mongodb.com/docs/atlas/atlas-search/path-construction/#std-label-ref-path">Path Construction</a>
+     */
+    fun path(path: String) {
+        document["path"] = path
+    }
+
+    /**
+     * The score assigned to matching search term results. Use one of the following options to modify the score:
+     *
+     * - boost: multiply the result score by the given number.
+     * - constant: replace the result score with the given number.
+     * - function: replace the result score using the given expression.
+     *
+     *  @param scoreConfiguration The configuration block for [ScoreSearchOptionDsl]
+     */
+    fun score(scoreConfiguration: ScoreSearchOptionDsl.() -> Unit) {
+        document["score"] = ScoreSearchOptionDsl().apply(scoreConfiguration).get()
+    }
+
+    /**
+     * Value or values to search.
+     * Value can be either a single value or an array of values of only one of the supported BSON types and can't be a mix of different types.
+     *
+     * @param value Value to search.
+     */
+    fun value(value: Boolean) {
+        document["value"] = value
+    }
+
+    /**
+     * Value or values to search.
+     * Value can be either a single value or an array of values of only one of the supported BSON types and can't be a mix of different types.
+     *
+     * @param value Values to search.
+     */
+    fun value(vararg value: Boolean) {
```

Comment:
I think it would be better to have a method of iterable arg version.

```kt
    /**
     * Value or values to search.
     * Value can be either a single value or an array of values of only one of the supported BSON types and can't be a mix of different types.
     *
     * @param value Values to search.
     */
    fun value(value: Iterable<Boolean>) {
        document["value"] = value
    }
```